### ISSUE TITLE
Able to enter day number to 39, month number to 19

### ISSUE THREAD

**Issue by ezhkov** [xdan/datetimepicker#81]
When the mask is enabled, user can enter from keyboard day number more than 31 (to 39), month number more than 12 (to 19). After that, date is calculating as in picture below
![datetimepicker](https://f.cloud.github.com/assets/568355/2491413/ab2b4cdc-b1eb-11e3-9640-5a117d2e0cee.png)

**Comment by slucas-nwx** [comment]
It would be nice to know the explanation behind this. I'm very curious to know what edge cases this applies to. Thanks. :)

**Comment by GGosseyn** [comment]
I encountered the same issue when using the moment.js parser, in order to be able to use another date formatter with a masked date input.
My hypothesis (I'm far from a specialist of this plugin) is that the mask syntax is very restrictive:
From my little experience, I think it allows the following rules when defining a mask (using JQuery.datetimepicker.setDateFormatter({... formatMask: function(value) { ... })):

- A '9' represents any digit,
- A digit D which value is **lower than 9** represents any digit value lower or equal to this digit value D
- A 'm' represents a digit or a letter
- A '*' represents any character,
- A specific character stays 'as is'

I have a similar implementation using moment.js like the one given in [this XDSoft example](https://xdsoft.net/jqplugins/datetimepicker/#use_other_date_parser).
In this example, I think that the formatMask is used to convert the date format into a mask definition.
So, for instance, an ISO 8601 date format like mine (moment.js syntax): 'YYYY-MM-DDTHH:mm:ss'  is converted into '9999-19-39T29:59:59' (XDSoft syntax equivalent should be 'Y-m-dTH:i:s' I think).

Here, you can see that the day of month is represented by 39, the month is represented by 19, and the hour in the day is represented by 29.
If one defines the month with 12, it would restrict the second digit to be 2 at most, preventing tor represented months 03,04,05,06,07,08 and 09 in which the second digit is >2.
The same problem occurs with the day of month which can go over 31 up to 39 and the hour in the day which can go over 24 up to 29.

In conclusion, I think that the mask definition syntax is not powerful enough and would greatly benefit to directly use the date format as a mask. 
So, the ISO 8601 mask up here could be defined like:

- 'Y-m-dTH:i:s' (using the XDSoft date format syntax) or alternatively
- 'YYYY-MM-DDTHH:mm:ss' (using the moment.js date format syntax)
instead of the currently used '9999-19-39T29:59:59' syntax.
